### PR TITLE
learn2016: filter code refactor to be a bit easier to read.

### PR DESCRIPTION
**Pull request by breville**

### CODE REVIEW

**Comment by breville** [comment]
I think this is a slight improvement?

File: apps/src/tutorialExplorer/tutorialSet.jsx [hunk 1]
```diff
@@ -63,40 +63,43 @@ const TutorialSet = React.createClass({
...
         }
 
-        return !filterGroupMiss;
+        return filterGroupsSatisfied;
       });
+    },
+
+    // Given a filter group, and the tutorial's relevant tag for that filter group,
+    // see if there's at least a single match.
+    findMatchingTag(filterGroup, tutorialTags) {
```
Comment (Bjvanminnen) [inline apps/src/tutorialExplorer/tutorialSet.jsx:94]: I might add some @param/@return comments to specify what we expect the input/output data to look like.

Also, I think we can maybe simplify this further

`return filterGroup.some(filterName => tutorialTagsSplit.includes(filtername))`

File: apps/src/tutorialExplorer/tutorialSet.jsx [hunk 2]
```diff
@@ -63,40 +63,43 @@ const TutorialSet = React.createClass({
...
-                filterHit = true;
-              }
-            }
-
-            // The filter group needs at least one user-selected filter to hit
-            // on the tutorial.
-            if (filterGroup.length !== 0 && !filterHit) {
-              filterGroupMiss = true;
+          if (filterGroup.length !== 0 && tutorialTags && tutorialTags.length > 0) {
+            if (!TutorialSet.findMatchingTag(filterGroup, tutorialTags)) {
```
Comment (Bjvanminnen) [inline apps/src/tutorialExplorer/tutorialSet.jsx]: Could just have this as another && instead of adding another level of nesting.

**Comment by Bjvanminnen** [comment]
lgtm

**Pull request merged**